iphone/Maps/Classes/Search/MWMSearch.mm

```diff
@@ -50,56 +51,76 @@ - (instancetype)initManager
 {
   self = [super init];
   if (self)
-  {
     _observers = [TObservers weakObjectsHashTable];
-    __weak auto weakSelf = self;
-    m_params.m_onStarted = [weakSelf] {
-      __strong auto self = weakSelf;
-      if (!self)
-        return;
-      runAsyncOnMainQueue(^{
-        [self onSearchStarted];
-      });
-    };
-    m_params.m_onResults = [weakSelf](search::Results const & results) {
-      __strong auto self = weakSelf;
-      if (!self)
-        return;
-      runAsyncOnMainQueue([self, results] {
-        if (results.IsEndMarker())
-        {
-          [self onSearchCompleted];
-        }
-        else
-        {
-          self->m_results = results;
-          self.suggestionsCount = results.GetSuggestsCount();
-          self.resultsCount = results.GetCount();
-          [self onSearchResultsUpdated];
-        }
-      });
-    };
-  }
   return self;
 }
 
+- (void)updateCallbacks
+{
+  __weak auto weakSelf = self;
+  m_everywhereParams.m_onResults = [weakSelf](search::Results const & results) {
+    __strong auto self = weakSelf;
+    if (!self)
+      return;
+    if (results.IsEndMarker())
+    {
+      self.everywhereSearchActive = NO;
+      [self onSearchCompleted];
+    }
+    else
+    {
+      self->m_results = results;
```

Comment:
onResults callback may be called multiple times. Is it OK that results is assigned, not appended to m_results?